baggage-processor/src/main/java/io/opentelemetry/contrib/baggage/processor/BaggageSpanProcessor.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.baggage.processor;
+
+import io.opentelemetry.api.baggage.Baggage;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.trace.ReadWriteSpan;
+import io.opentelemetry.sdk.trace.ReadableSpan;
+import io.opentelemetry.sdk.trace.SpanProcessor;
+import java.util.function.Function;
+
+/**
+ * This span processor copies attributes stored in {@link Baggage} into each newly created {@link
+ * io.opentelemetry.api.trace.Span}.
+ */
+public class BaggageSpanProcessor implements SpanProcessor {
+  private Function<String, Boolean> keyFilter = key -> true;
+
+  /**
+   * Creates a new {@link BaggageSpanProcessor} that copies all baggage entries into the newly
+   * created {@link io.opentelemetry.api.trace.Span}.
+   */
+  public BaggageSpanProcessor() {}
+
+  /**
+   * Creates a new {@link BaggageSpanProcessor} that copies only baggage entries with keys that pass
+   * the provided filter into the newly created {@link io.opentelemetry.api.trace.Span}.
+   */
+  public BaggageSpanProcessor(Function<String, Boolean> keyFilter) {
```

Comment:
I would recommend making the filter a [BiPredicate<String,BaggageEntry>](https://docs.oracle.com/javase/8/docs/api/java/util/function/BiPredicate.html). The entry can be ignored, but maybe someone might want to filter on the contents of the baggage.

Reply:
(a BiPredicate constructor could always be added later)
```suggestion
  public BaggageSpanProcessor(Predicate<String> keyFilter) {
```

Reply:
Added in https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1290/commits/331f2e2de9429a59333c538950c81f9b9e130869